From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: vantoman <Mustafa.vantom@gmail.com>
Date: Sun, 4 Jun 2023 13:39:51 +0200
Subject: [PATCH] ipa_v3: Fix enum conversion warnings

../techpack/dataipa/drivers/platform/msm/ipa/ipa_v3/rmnet_ipa.c:510:41: warning: implicit conversion from enumeration type 'enum ipa_ip_type_enum_v01' to different enumeration type 'enum ipa_ip_type' [-Wenum-conversion]
        q6_ul_flt_rule_ptr->ip = flt_spec_ptr->ip_type;
                               ~ ~~~~~~~~~~~~~~^~~~~~~
../techpack/dataipa/drivers/platform/msm/ipa/ipa_v3/rmnet_ipa.c:511:45: warning: implicit conversion from enumeration type 'enum ipa_filter_action_enum_v01' to different enumeration type 'enum ipa_flt_action' [-Wenum-conversion]
        q6_ul_flt_rule_ptr->action = flt_spec_ptr->filter_action;
                                   ~ ~~~~~~~~~~~~~~^~~~~~~~~~~~~

Change-Id: I0eb68d707151cd103676a30659ab81bf6fced131
Signed-off-by: Sevenrock <sevenrock@hotmail.de>
---
 techpack/dataipa/drivers/platform/msm/ipa/ipa_v3/rmnet_ipa.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/techpack/dataipa/drivers/platform/msm/ipa/ipa_v3/rmnet_ipa.c b/techpack/dataipa/drivers/platform/msm/ipa/ipa_v3/rmnet_ipa.c
index 90f04cc6b3f2..44db74269dfa 100644
--- a/techpack/dataipa/drivers/platform/msm/ipa/ipa_v3/rmnet_ipa.c
+++ b/techpack/dataipa/drivers/platform/msm/ipa/ipa_v3/rmnet_ipa.c
@@ -509,8 +509,8 @@ static void ipa3_copy_qmi_flt_rule_ex(
 	 */
 	flt_spec_ptr = (struct ipa_filter_spec_ex_type_v01 *) flt_spec_ptr_void;
 
-	q6_ul_flt_rule_ptr->ip = flt_spec_ptr->ip_type;
-	q6_ul_flt_rule_ptr->action = flt_spec_ptr->filter_action;
+	q6_ul_flt_rule_ptr->ip = (enum ipa_ip_type)flt_spec_ptr->ip_type;
+	q6_ul_flt_rule_ptr->action = (enum ipa_flt_action)flt_spec_ptr->filter_action;
 	if (flt_spec_ptr->is_routing_table_index_valid == true)
 		q6_ul_flt_rule_ptr->rt_tbl_idx =
 		flt_spec_ptr->route_table_index;
-- 
2.39.5

